iceoryx_posh/source/roudi/port_manager.cpp

```diff
@@ -764,7 +764,23 @@ PortManager::acquirePublisherPortData(const capro::ServiceDescription& service,
                                       const RunnableName_t& runnable [[gnu::unused]], // @todo #25 Fix introspection
                                       const PortConfigInfo& portConfigInfo) noexcept
 {
-    /// @todo #25 Add unique check for single producer type - see acquireSenderPortData()
+#if defined(RESTRICT_TO_1_TO_N_COMMUNICATION)
+    // check if the publisher is already in the list
+    for (auto publisherPortData : m_portPool->getPublisherPortDataList())
+    {
+        popo::PublisherPortRouDi publisherPort(publisherPortData);
+        if (service == publisherPort.getCaProServiceDescription())
+        {
+            LogWarn() << "Process '" << processName
+                      << "' tried to register an unique PublisherPort which is already used by '"
+                      << publisherPortData->m_processName << "' with service '"
+                      << service.operator cxx::Serialization().toString() << "'.";
+
+            errorHandler(Error::kPOSH__PORT_MANAGER_PUBLISHERPORT_NOT_UNIQUE, nullptr, ErrorLevel::MODERATE);
+            return cxx::error<PortPoolError>(PortPoolError::UNIQUE_PUBLISHER_PORT_ALREADY_EXISTS);
```

Comment:
Is it necessary to call the error handler here? We already return an error that is sent via mq and finally evaluated in the runtime.
I guess our error handling needs to be revised anyway.

Reply:
I suppose that's part of #130

Reply:
This will be an error reported by RouDi, the one sent over the message queue will be in the user application. So we somehow log here that an application does not act according to the contract. I would like to have the moderate error here